### ISSUE TITLE
`multiline-comment-style`: auto fixing for `starred-block` doesn't work for `jsdoc` tags

### ISSUE THREAD

**Issue by JounQin** [eslint-stylistic/eslint-stylistic#757]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

`starred-block` auto fixing should start with `/**` instead of `/*`

### Reproduction

https://stackblitz.com/edit/github-y8d4pnky?file=src%2Findex.ts

### Contributes

- [x] I am willing to submit a PR to fix this issue
- [x] I am willing to submit a PR with failing tests

**Comment by JounQin** [comment]
The following comment autofixing is also strange

Input:

```js
/*
  Copyright © 2019 Andrew Powell
  This Source Code Form is subject to the terms of the Mozilla Public
  License, v. 2.0. If a copy of the MPL was not distributed with this
  file, You can obtain one at http://mozilla.org/MPL/2.0/.
  The above copyright notice and this permission notice shall be
  included in all copies or substantial portions of this Source Code Form.
*/
```

Output:

```js
/*
 *Copyright © 2019 Andrew Powell
 *This Source Code Form is subject to the terms of the Mozilla Public
 *License, v. 2.0. If a copy of the MPL was not distributed with this
 *file, You can obtain one at http://mozilla.org/MPL/2.0/.
 *The above copyright notice and this permission notice shall be
 *included in all copies or substantial portions of this Source Code Form.
 */
```

Expected:

```js
/**
 * Copyright © 2019 Andrew Powell
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 * The above copyright notice and this permission notice shall be
 * included in all copies or substantial portions of this Source Code Form.
 */
```

**Comment by JounQin** [comment]
It should ignore triple slash references:

```ts
/// <reference types="vite/client" />
/// <reference types="vitest/globals" />
```

workaround: add a new line between they

```ts
/// <reference types="vite/client" />

/// <reference types="vitest/globals" />
```
